xla/pjrt/c/pjrt_c_api_custom_partitioner_extension.h

```diff
@@ -79,32 +79,32 @@ struct JAX_CustomCallPartitioner_InferShardingFromOperands_Args {
 
   bool has_result_sharding;
   JAX_CustomCallPartitioner_string result_sharding;
-};
+} JAX_CustomCallPartitioner_InferShardingFromOperands_Args;
 
-struct JAX_CustomCallPartitioner_PropagateUserSharding_Args {
+typedef struct JAX_CustomCallPartitioner_PropagateUserSharding_Args {
   JAX_CustomCallPartitioner_version_and_error header;
 
   JAX_CustomCallPartitioner_string backend_config;
 
   JAX_CustomCallPartitioner_string result_shape;
 
   JAX_CustomCallPartitioner_string result_sharding;  // inout
-};
+} JAX_CustomCallPartitioner_PropagateUserSharding_Args;
 
-struct JAX_CustomCallPartitioner_Callbacks {
+typedef struct JAX_CustomCallPartitioner_Callbacks {
   int64_t version;
   void* private_data;
-  void (*dtor)(JAX_CustomCallPartitioner_Callbacks* data);
-  void (*partition)(JAX_CustomCallPartitioner_Callbacks* data,
+  void (*dtor)(struct JAX_CustomCallPartitioner_Callbacks* data);
```

Comment:
Do you need `struct` in the arguments if you added the `typedefs` above? (here and below)